test/unit/Dataflow_utils.ml

```diff
@@ -420,7 +420,7 @@ let example4_program =
       }
       |}
   in
-  let mir = Ast_to_Mir.trans_prog "" (semantic_check_program ast) in
+  let mir = Ast_to_Mir.trans_prog "" (fst @@ semantic_check_program ast) in
```

Comment:
We should write a new function that exposes the additional API and leave the old one alone instead of adding `fst` to everything.

Reply:
How about the tests to expect no warnings; semantic check always returns a (possibly empty) list of warnings with either success (a typed AST) or failure (a non-empty list of errors) so it makes sense to test that.

Reply:
Added explicit expect tests for warnings